### PR TITLE
Checks for broken links

### PR DESCRIPTION
**Pull request by vegaro**
Adds a link to check for links that 404

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -204,6 +205,13 @@ lane :migration do |options|
     clean_docs_source_headers
 end
 
+lane :check_links do |options|
+    errors = check_links_in_folder(SOURCE_FOLDER)
```
Comment (NachoSoto) [inline fastlane/Fastfile:209]: <3

**Review by NachoSoto — APPROVED** [review]
This is awesome <3

File: fastlane/checks.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+def check_links_in_file(markdown_file)
+    markdown_content = get_file_contents(markdown_file)
+    links = markdown_content.scan(/\[([^\]]+)\]\(([^)]+)\)/)
+
+    errors = []
+    links.each do |text, url|
+        next unless url.start_with?('http')
+
+        Fastlane::UI.message("Checking link: #{url}")
```
Comment (tonidero) [inline fastlane/checks.rb:20]: I wonder if these messages might be too noisy... We can leave it for now though and remove them later if needed.

**Review by tonidero — APPROVED** [review]
Nice check! It looks like it already found some broken links ❤️

File: fastlane/checks.rb (hunk 2 above)
Comment (vegaro) [reply fastlane/checks.rb:20]: True, I will remove it, the link will be printed in the success/error

**Comment by vegaro** [comment]
Changed the approach to use redcarpet to render the markdown into HTML and parse the `a` links instead. I think it's more solid. I was getting issues with the regex and urls that contain `)` in them, like: `https://revenuecat.github.io/purchases-ios-docs/4.19.0/documentation/revenuecat/purchases/customerinfo(fetchpolicy:)/`

This approach is only ~10 seconds slower

**Pull request merged**